### PR TITLE
feat: add validator ui support for pipeline templates

### PR DESCRIPTION
**Pull request by pritamstyz4ever**
## Context

Added validator support to display pipeline template and pipeline template usage

## Objective

The validator should correctly display the pipeline template name, namespace, jobs, etc.
The validator should correctly display the pipeline template used in a pipeline

Template:
Screenshot 2024-03-06 at 7 17 55 PM
Screenshot 2024-03-06 at 7 18 03 PM

Usage:
Screenshot 2024-03-07 at 2 41 11 PM

## References
https://github.com/screwdriver-cd/screwdriver/issues/3052

## License

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

**Review by jithine — DISMISSED** [review]

File: tests/integration/components/validator-results/component-test.js [hunk 1]
```diff
@@ -110,6 +110,37 @@ module('Integration | Component | validator results', function (hooks) {
...
+    this.set('validationMock', {
+      errors: [],
+      template: {
+        namespace: 'batman',
+        name: 'batmobile',
+        version: '1.0.0',
+        config: {
+          shared: {
+            image: 'int-test:1'
+          },
```
Comment (sagar1312) [inline tests/integration/components/validator-results/component-test.js:123]: We should use/render the configuration under `shared`.
If few jobs inherit `steps`, `settings` from shared, that is not being visualized in the validator results

**Review by sagar1312 — CHANGES_REQUESTED** [review]

**Comment by sagar1312** [comment]
Also the workflow is not visualized for pipeline template
We should show how the jobs are interconnected in the workflow graph

image

**Comment by pritamstyz4ever** [comment]
> Also the workflow is not visualized for pipeline template We should show how the jobs are interconnected in the workflow graph
> 
> image

Current workflow jobs are not returned by the API for templates. We will need to add that

File: tests/integration/components/validator-results/component-test.js (hunk 1 above)
Comment (pritamstyz4ever) [reply tests/integration/components/validator-results/component-test.js:123]: I think shared steps have the same behavior as pipeline, for settings and environment we may need to tweak API to return the merged values

**Comment by sagar1312** [comment]
> > Also the workflow is not visualized for pipeline template We should show how the jobs are interconnected in the workflow graph
> > image
> 
> Currently workflow jobs are not returned by the API for templates. We will need to add that

Yes, the validator API may need to all the config-parser again to merge the `shared` into `jobs`. That would also generate workflow graph.

**Comment by VonnyJap** [comment]
@sagar1312 

> Also the workflow is not visualized for pipeline template
> We should show how the jobs are interconnected in the workflow graph

We can do the visualization as next item to work on. At least the validator can make sure that the config in template is working. Although yes, I agree, without the workflow graph, pipeline template validator is missing the important purpose and becomes mainly just to validate the syntax.

**Comment by sagar1312** [comment]
> @sagar1312
> 
> > Also the workflow is not visualized for pipeline template
> > We should show how the jobs are interconnected in the workflow graph
> 
> We can do the visualization as next item to work on. At least the validator can make sure that the config in template is working. Although yes, I agree, without the workflow graph, pipeline template validator is missing the important purpose and becomes mainly just to validate the syntax.

Sure, we can push changes incrementally. 
Besides visualizing workflow graph, we also need to make sure config under `shared` as explained earlier needs to be somehow (merging it with `jobs`) displayed in the result.

**Review by sagar1312 — APPROVED** [review]

**Review by sagar1312 — APPROVED** [review]

**Comment by adong** [comment]
Validator: https://cd.screwdriver.cd/validator

```
template: sd-test/example-template@latest
shared:
    environment: 
        FOO: bar
```

**Pull request merged**